### PR TITLE
firewalk: init at 5.0

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -42344,4 +42344,6 @@ with pkgs;
   ssl-proxy = callPackage ../tools/networking/ssl-proxy { };
 
   code-maat = callPackage ../development/tools/code-maat {};
+
+  firewalk = callPackage ../applications/networking/firewalk {};
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:42348]:
```suggestion
```

File: pkgs/applications/networking/firewalk/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib
```
Comment (AndersonTorres) [inline pkgs/applications/networking/firewalk/default.nix:1]: Move this file to `pkgs/by-name/fi/firewalk/package.nix`.

File: pkgs/applications/networking/firewalk/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+, libpcap
+, libdnet
+}:
+
+stdenv.mkDerivation rec {
+  pname = "firewalk";
+  version = "5.0";
+
+  src = fetchurl {
+    url = "https://salsa.debian.org/pkg-security-team/firewalk/-/archive/debian/5.0-5/firewalk-debian-5.0-5.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/applications/networking/firewalk/default.nix:14]: Why are you using a Debian file?

File: pkgs/applications/networking/firewalk/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "luFTi4DnXbiNDI2uPgbtF4Y9m/SdxQBs+Dfdmck7mFU=";
```
Comment (AndersonTorres) [inline pkgs/applications/networking/firewalk/default.nix:15]:
```suggestion
    hash = "sha256-luFTi4DnXbiNDI2uPgbtF4Y9m/SdxQBs+Dfdmck7mFU=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/firewalk/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  buildInputs = [ libnet libpcap libdnet ];
+
+  meta = with lib; {
+    description = "Gateway ACL scanner";
+    homepage = "http://packetfactory.openwall.net/projects/firewalk/";
+    platforms = platforms.linux;
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ tochiaha ];
```
Comment (AndersonTorres) [inline pkgs/applications/networking/firewalk/default.nix:25]:
```suggestion
    maintainers = with maintainers; [ ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by 999eagle** [comment]
This supersedes #260590. @Tochiaha it's not really helpful if you repeatedly close and re-open PRs like this.

**Comment by Tochiaha** [comment]
@999eagle Thank you for your feedback on the continuous closing and opening of PR, i will make sure it stops from now on.

**Review by AndersonTorres — APPROVED** [review]
Ping me 24h to merge

**Review by 999eagle — APPROVED** [review]
Result of `nixpkgs-review pr 261547` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>firewalk</li>
  </ul>
</details>

File: pkgs/applications/networking/firewalk/default.nix (hunk 5 above)
Comment (fabaff) [reply pkgs/applications/networking/firewalk/default.nix]: New packages need a maintainer -> https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-packages

File: pkgs/by-name/fi/firewalk/package.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{ lib
+, stdenv
+, fetchurl
+, libnet
+, libpcap
+, libdnet
+}:
+
+stdenv.mkDerivation rec {
```
Comment (wegank) [inline pkgs/by-name/fi/firewalk/package.nix:9]:
```suggestion
stdenv.mkDerivation {
```

File: pkgs/by-name/fi/firewalk/package.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+, libpcap
+, libdnet
+}:
+
+stdenv.mkDerivation {
+  pname = "firewalk";
+  version = "5.0";
+
+  src = fetchurl {
+    url = "https://salsa.debian.org/pkg-security-team/firewalk/-/archive/upstream/5.0/firewalk-upstream-5.0.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/by-name/fi/firewalk/package.nix:14]: Parameterize `version` here.


```suggestion
    url = "https://salsa.debian.org/pkg-security-team/firewalk/-/archive/upstream/${finalAttrs.version}/firewalk-upstream-${finalAttrs.version}.tar.gz";
```

File: pkgs/by-name/fi/firewalk/package.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{ lib
+, stdenv
+, fetchurl
+, libnet
+, libpcap
+, libdnet
+}:
+
+stdenv.mkDerivation {
```
Comment (AndersonTorres) [inline pkgs/by-name/fi/firewalk/package.nix:9]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/fi/firewalk/package.nix [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  buildInputs = [ libnet libpcap libdnet ];
+
+  meta = with lib; {
+    description = "Gateway ACL scanner";
+    homepage = "http://packetfactory.openwall.net/projects/firewalk/";
+    platforms = platforms.linux;
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ tochiaha ];
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/by-name/fi/firewalk/package.nix:27]:
```suggestion
})
```

**Review by AndersonTorres — DISMISSED** [review]

**Pull request merged**